File: solutions/48/sol.js
const MOD = 10_000_000_000
const LIMIT = 1000;

const pow = (n) => {
  let result = 1;
  for (let index = 0; index < n; index++) {
    result = (result * n) % MOD;
  }
  return result;
}

const getAnswer = (n) => {
  let answer = 0;
  for (let num = 1; num <= n; num++) {
    answer = (answer + pow(num)) % MOD;
  }
  return answer;
}

const answer = getAnswer(LIMIT);
console.log(answer);

File: solutions/48/sol.go
package main

import (
	"fmt"
	"math"
)

var K = int(math.Pow(10, 10))

func getProduct(n int) int {
	product := 1
	for i := 0; i < n; i++ {
		product = (product * n) % K
	}
	return product
}

func getAnswer(n int) int {
	answer := 0
	for i := 1; i <= n; i++ {
		answer = (answer + getProduct(i)) % K
	}
	return answer
}

func main() {
	fmt.Println(getAnswer((10)))
	fmt.Println(getAnswer((1000)))
}
